pkg/controllers/cleanup/controller.go

```diff
@@ -182,6 +182,26 @@ func (c *controller) cleanup(ctx context.Context, logger logr.Logger, policy kyv
 	debug := logger.V(4)
 	var errs []error
 
+	// get the deletion policy
+	deletionPolicy := spec.DeletionPropagationPolicy
+
+	// Add delete options based on the deletion policy
+	deleteOptions := &metav1.DeleteOptions{
+		PropagationPolicy: deletionPolicy,
+	}
+	// Check if the deletion policy is not provided, and set to default if needed
+	if deletionPolicy != nil {
+		// If the user has provided a deletion policy, use it
+		deleteOptions.PropagationPolicy = deletionPolicy
+		logger.Info("Using specified deletion propagation policy", "policy", *deletionPolicy)
+	} else {
+		// If the deletion policy is not provided, you can either
+		// set it to a default value
+		defaultPolicy := metav1.DeletePropagationForeground // Setting default as Foreground
+		deleteOptions.PropagationPolicy = &defaultPolicy
+		logger.Info("DeletionPropagationPolicy not provided, setting to default (Foreground)")
+	}
+
```

Comment:
Why do we need this ?

Reply:
We don't want to default, if nothing was specified let the apiserver decide.

Reply:
Okay . Let me change the logic then.